src/entry.js

```diff
@@ -43,23 +43,25 @@ class Entry {
       clock: new Clock(clockId, clockTime),
     }
 
-    // If signing key was passedd, sign the enrty
-    if (keystore && signKey) {
-      entry = await Entry.signEntry(keystore, entry, signKey) 
-    }
-
+    const signature = await identity.provider.sign(encode(entry))
+    entry.key = identity.publicKey
```

Comment:
In order to make it possible to verify that the signing key belongs to the identity that the entry claims to have, the entry key needs to contain a) the original ID of the user (identity.id) b) the public signing key (identity.publicKey) and c) a signature of <id + publicKey> signed by the original ID (eg. an eth address).

I believe a good way to do that would be to add a function to `Identity` class and return all 3 in one object, like:
```javascript
toJSON () {
  return {
    id: this.id,
    publicKey: this.publicKey,
    signature: this.signature
  }
}
```

Which then would mean that this line then would become `entry.key = identity.toJSON()`.